### ISSUE TITLE
Custom Shaders

### ISSUE THREAD

**Issue by GoddessServant0** [ififfy/flipflip#176]
https://www.reddit.com/r/EroticHypnosis/comments/cegsty/flipflip_v230_hypno_slideshow_multiple_windows/ev8v6n1?utm_source=share&utm_medium=web2x

> Is there any kind of support for custom effects? I've got a number of glsl shades written for effects that I'd love to be able to overlay.
> I use a program called [Shadron](http://www.arteryengine.com/shadron/) which uses a c-like scripting language to help glue everything together. Another popular tool for writing glsl code is [Shadertoy](https://www.shadertoy.com/view/lds3RH).
> Here's a pretty simple one that makes the image "ripple" ([example](https://i.imgur.com/hXle5pZ.png)): [ripple.shadron](https://www.dropbox.com/s/r85ncxh1ok885k0/ripple.shadron?dl=0)
> And here's a somewhat more complex one that applies various methods of edge detection ([example](https://i.imgur.com/Ko0KVvj.png)): [edge.shadron](https://www.dropbox.com/s/iuwulpx7e1v4zz4/prewitt_edge_spiral.shadron?dl=0) and [lib/edge_detect.shadron](https://www.dropbox.com/s/ap8hdym9aj5vfx9/edge_detect.shadron?dl=0)

**Comment by ififfy** [comment]
I don't think this is possible with browser tech. Shaders run in WebGL views, where all rendering is super low level. You can't just do stuff to a video stream with it, unfortunately.

**Comment by GoddessServant0** [comment]
You can apply WebGL shaders to an image object by rendering on a canvas, however. Longshot idea, but just put it here as a request.

**Comment by ififfy** [comment]
Yeah...rendering video onto a canvas isn't going to work, or at least it'll guzzle CPU like a ___ guzzles ___.

**Comment by abrenoch** [comment]
Just FYI, the [three.js post processing library](https://github.com/pmndrs/postprocessing) seems to work fine with videos rendered to the canvas. I have been experimenting with it in a webXR video player and seems very viable.
